### PR TITLE
Improve docs on basetemp and retention

### PR DESCRIPTION
**Pull request by soxofaan**
Improve coverage of current handling of `--basetemp` option and its lack of retention functionality.

- ~[ ] Include documentation when adding new features.~
- ~[ ] Include new tests or update existing tests when applicable.~
- [X] Allow maintainers to push and squash when merging my commits.
- [x] Add text like ``closes #XYZW`` to the PR description and/or commits (where ``XYZW`` is the issue number). See the [github docs](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) for more information.
- [x] Create a new changelog file in the `changelog` folder, with a name like `<ISSUE NUMBER>.<TYPE>.rst`. See [changelog/README.rst](https://github.com/pytest-dev/pytest/blob/main/changelog/README.rst) for details.
- [x] Add yourself to `AUTHORS` in alphabetical order.

Addresses (part of) issue #10829:
the `--basetemp` option and the default retention feature can not be combined currently. This is completely unclear from the current docs. While I initially requested to allow the combination of retention with `--basetemp` in #10829, this PR just attempts better document the current behavior.

### CODE REVIEW

**Comment by soxofaan** [comment]
Note that this PR introduces the exposure of the `PYTEST_DEBUG_TEMPROOT` env var
https://github.com/pytest-dev/pytest/blob/26215b8f795a2b13acc1a8827fdfb6a29b502a6a/src/_pytest/tmpdir.py#L157-L158

I'm not sure if that is desired. Maybe it's only intended as internal implementation detail for testing purposes.

However, I think it's pretty valuable to have this a tool to finetune the "temproot" in practice (e.g. as discussed in #10829 as well)

File: doc/en/how-to/tmp_path.rst [hunk 1]
```diff
@@ -133,27 +133,47 @@ API for details.
...
+The temporary directories,
+as returned by the :fixture:`tmp_path` and (now deprecated) :fixture:`tmpdir` fixtures,
+are automatically created under a base temporary directory,
+in a structure that depends on the ``--basetemp`` option:
 
-Using the ``--basetemp``
-option will remove the directory before every run, effectively meaning the temporary directories
-of only the most recent run will be kept.
+- By default (when the ``--basetemp`` option is not set),
+  the temporary directories will follow this this template:
```
Comment (nicoddemus) [inline doc/en/how-to/tmp_path.rst:142]:
```suggestion
  the temporary directories will follow this template:
```

**Review by nicoddemus — APPROVED** [review]
This is looks much better, thanks!

File: doc/en/how-to/tmp_path.rst [hunk 2]
```diff
@@ -133,27 +133,47 @@ API for details.
...
-.. code-block:: bash
+      {temproot}/pytest-of-{user}/pytest-{num}/{testname}/
 
-    pytest --basetemp=mydir
+  where:
 
-.. warning::
+  - ``{temproot}`` is the system temporary directory
+    as determined by :py:func:`tempfile.gettempdir`.
+    It can be overridden by the ``PYTEST_DEBUG_TEMPROOT`` environment variable.
```
Comment (nicoddemus) [inline doc/en/how-to/tmp_path.rst:152]: We might as well document this in `reference.rst`, and reference it from here using `:envvar:PYTEST_DEBUG_TEMPROOT`.

**Comment by soxofaan** [comment]
> Note that this PR introduces the exposure of the PYTEST_DEBUG_TEMPROOT env var

a concern here is that this env var contains "DEBUG" in it's name, which makes it look like it's not ready for production use. As a user I'm not interested to use this for one-off debug use cases, I want to use this in real CI contexts.

Is there opportunity to rename this env var to, for example, `PYTEST_TEMPROOT` ?

**Comment by nicoddemus** [comment]
Regardless if we want to rename or not, we should document the existing var -- renaming it should be discussed in a separate issue.

**Comment by soxofaan** [comment]
I'm suggesting the rename here as it seems less cumbersome  to do a  rename before it making a publicly documented feature than the other way around.

**Comment by soxofaan** [comment]
FYI I'm just assuming `PYTEST_DEBUG_TEMPROOT` is an undocumented implementation detail, as I couldn't find any pointers in public online  docs. But maybe there is already usage in the wild

**Comment by nicoddemus** [comment]
> FYI I'm just assuming PYTEST_DEBUG_TEMPROOT is an undocumented implementation detail, as I couldn't find any pointers in public online docs. But maybe there is already usage in the wild

You are right in your assumption, but it seems it is used in the wild:

https://github.com/search?q=PYTEST_DEBUG_TEMPROOT&type=code

So we do need to document it regardless, even if we were to change it later (which I don't think is something worthwhile to do).

**Comment by nicoddemus** [comment]
Thanks @soxofaan, looks great!

I will leave it open for a few more days to give others a chance to chime in. :+1:

**Pull request merged**